### PR TITLE
Bjs/response text modal

### LINKED ISSUE #307: Weigh In Response Modal
On non-mobile screen widths, clicking on a weigh-in text response should open as a modal overlay, showing the full response text. Upper right corner large X should close the modal. Background should be clickable to close the modal.

### PR DESCRIPTION
**Pull request by smithWEBtek**
Resolves #307 Weigh In Response Modal.

This change was necessary to meet design requirements.

This change provides the javascript and styles required to produce clickable modals with overlays.

The current side effect is that the response texts no longer have links to their original show pages.
All response texts (non-line-clamped and line-clamped) now go to modals on click, in all screen widths.

### CODE REVIEW

**Comment by mzagaja** [comment]
Could you rebase/squash these WIP commits before I review?

**Comment by smithWEBtek** [comment]
> Could you rebase/squash these WIP commits before I review?

Done

**Comment by mzagaja** [comment]
This seems to work generally but I am noticing a few behaviors that we might want to review:

1. The MetroCommon x 2050 header appears in front of the modal instead of behind it at certain sizes.
image

2. If the text is not long enough to trigger a modal, we still get a click cursor. Can we add a conditional to test for this?

File: app/assets/stylesheets/partials/_response_modal.scss [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  &-overlay {
+    background-color: rgba(63, 60, 60, 0.705);
+    height: 100%;
+    left: 0;
+    overflow: hidden;
+    position: fixed;
+    top: 0;
+    width: 100%;
+    z-index: 2;
```
Comment (mzagaja) [inline app/assets/stylesheets/partials/_response_modal.scss]: If you set this to `101` it should fix the header overlap problem.

File: spec/system/hero_rotation_spec.rb [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
     create(:hero_image, image: build(:alternate_image))
     Capybara.current_session.driver.browser.manage.window.resize_to(1440, 1000)
     visit "/"
+    # puts page.find(:xpath, '//*[@id="page"]/div/section[1]/div[1]/div[1]')['innerHTML']
     sleep 5
+    puts page.find(:xpath, '//*[@id="page"]/div/section[1]/div[1]/div[1]')['innerHTML']
     expect(page).to have_xpath('//div[contains(@style,"beach-alternate.jpeg")]')
   end
-
+
```
Comment (mzagaja) [inline spec/system/hero_rotation_spec.rb]: A random line of spaces escaped! 😉

File: app/assets/stylesheets/partials/_response_modal.scss (hunk 1 above)
Comment (smithWEBtek) [reply app/assets/stylesheets/partials/_response_modal.scss]: Done

File: spec/system/hero_rotation_spec.rb (hunk 2 above)
Comment (smithWEBtek) [reply spec/system/hero_rotation_spec.rb]: Yes looks like a commented out section / copy of a test. Removed it.

**Comment by smithWEBtek** [comment]
*re: If the text is not long enough to trigger a modal, we still get a click cursor. Can we add a conditional to test for this?*

I think the plan was to have them all click open into modals, regardless of line clamping (or not), but let's check w/Ryan tomorrow.

**Pull request merged**